doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -151,27 +152,122 @@ public void updateDoiRequest(UUID publicationIdentifier, ApiUpdateDoiRequest api
 
         Publication publication = fetchPublicationByIdentifier(publicationIdentifier);
 
+
         DoiRequest updatedDoiRequest =
             doiRequestCloneWithNewStatusAndNewMessage(publication, apiUpdateDoiRequest, requestedByUsername);
 
         replaceDoiRequestInPublication(publication, updatedDoiRequest);
         putItem(publication);
     }
 
+
+    @Override
+    public void addMessage(UUID publicationIdentifier, String message, UserInstance user)
+        throws ApiGatewayException {
+        Instant now = clockForTimestamps.instant();
+        Publication publication = fetchPublicationByIdentifier(publicationIdentifier);
+        authorizeSendingMessage(publication, user);
```

Comment:
Do we want a new class for the authorizer so we can hide away all these rules from the service?